ResearchKit/Common/ORKFormItemCell.m

```diff
@@ -1148,3 +1156,115 @@ - (BOOL)textField:(UITextField *)textField shouldChangeCharactersInRange:(NSRang
+    _selectionView.delegate = self;
+    
+    [self.contentView addSubview:_selectionView];
+
+    if (self.formItem.placeholder != nil) {
+        [_selectionView setPlaceholderText:self.formItem.placeholder];
+    }
+    
+    [self setUpConstraints];
+}
+
+- (void)setUpConstraints {
+    NSMutableArray *constraints = [NSMutableArray new];
+    
+    NSDictionary *dictionary = @{@"_selectionView":_selectionView};
+    ORKEnableAutoLayoutForViews([dictionary allValues]);
+    NSDictionary *metrics = @{@"verticalMargin":@(10), @"horizontalMargin":@(self.separatorInset.left), @"verticalMarginBottom":@(10.0 - (1.0 / [UIScreen mainScreen].scale))};
+    
+    [constraints addObjectsFromArray:[NSLayoutConstraint constraintsWithVisualFormat:@"H:|-horizontalMargin-[_selectionView]-horizontalMargin-|" options:NSLayoutFormatDirectionLeadingToTrailing metrics:metrics views:dictionary]];
+    [constraints addObjectsFromArray:[NSLayoutConstraint constraintsWithVisualFormat:@"V:|-verticalMargin-[_selectionView]-verticalMarginBottom-|" options:NSLayoutFormatDirectionLeadingToTrailing metrics:metrics views:dictionary]];
+    _heightConstraint = [NSLayoutConstraint constraintWithItem:_selectionView attribute:NSLayoutAttributeHeight relatedBy:NSLayoutRelationEqual toItem:nil attribute:NSLayoutAttributeNotAnAttribute multiplier:1.0 constant:_selectionView.intrinsicContentSize.height];
+    _heightConstraint.priority = UILayoutPriorityDefaultHigh;
+    [constraints addObject:_heightConstraint];
+    
+    [self.contentView addConstraints:constraints];
+}
+
+- (void)setFormItem:(ORKFormItem *)formItem {
+    [super setFormItem:formItem];
+    
+    if (_selectionView) {
+        [_selectionView setPlaceholderText:formItem.placeholder];
+    }
+}
+
+- (void)answerDidChange {
+    _selectionView.answer = self.answer;
+}
+
+- (void)setEditingHighlight:(BOOL)editingHighlight {
+    _editingHighlight = editingHighlight;
+    [_selectionView setTextColor:( _editingHighlight ? [self tintColor] : [UIColor blackColor])];
+}
+
+- (void)locationSelectionViewDidBeginEditing:(ORKLocationSelectionView *)view {
+    self.editingHighlight = YES;
+    [_selectionView showMapView];
+    [self.delegate formItemCellDidBecomeFirstResponder:self];
+}
+
+- (void)locationSelectionViewDidEndEditing:(ORKLocationSelectionView *)view {
+    self.editingHighlight = NO;
+    [self.delegate formItemCellDidResignFirstResponder:self];
+}
+
+- (void)locationSelectionViewDidChange:(ORKLocationSelectionView *)view {
+    [self inputValueDidChange];
+}
+
+- (void)locationSelectionViewNeedsResize:(ORKLocationSelectionView *)view {
+    UITableView *tableView = [self parentTableView];
+    
+    _heightConstraint.constant = _selectionView.intrinsicContentSize.height;
+    
+    [tableView beginUpdates];
+    
+    CGRect convertedVisibleRect = [tableView convertRect:view.bounds fromView:view];
+    [tableView scrollRectToVisible:convertedVisibleRect animated:YES];
```

Comment:
Maybe you can make this `convertedVisibleRect` more focus on the text field instead of the map view